nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/util/JdbcCommon.java

```diff
@@ -284,7 +291,7 @@ public static Schema createSchema(final ResultSet rs, String recordName, boolean
                     break;
 
                 case INTEGER:
-                    if (meta.isSigned(i)) {
+                    if (!dbAdapter.getSupportsMetaDataColumnIsSigned() || meta.isSigned(i)) {
```

Comment:
If the DB doesn't support isSigned(), should we treat them all as Longs? I assume it's done this way if Hive returns INTEGER as the type and Integer objects for values. I wonder if there's another DB that doesn't support isSigned() and could return Long objects. In that case we might want another check for getSupportsMetaDataColumnIsSigned() (and for the type of object) when storing the values into the record.